### PR TITLE
Ignore cross-origin requests; Remove listener on detached cb

### PR DESCRIPTION
**Pull request by jmalonzo**

### CODE REVIEW

File: src/pushstate-anchor.js [hunk 1]
```diff
@@ -6,13 +6,14 @@
   var HTMLPushStateAnchorElement = Object.create(HTMLAnchorElement.prototype);
 
   function pushStateAnchorEventListener(event) {
-    // open in new tab
-    if (event.ctrlKey || event.metaKey || event.which === 2) {
+    var href = this.getAttribute('href');
+    // open in new tab or cross-origin requests
+    if (event.ctrlKey || event.metaKey || event.which === 2 || href.startsWith("http") && window.location.host !== new URL(href).host) {
```
Comment (erikringsmuth) [inline src/pushstate-anchor.js]: this all looks good, but can you split this into 2 separate if checks for clarity around the comments. have the first one check for opening in a new tab and the second one can check if the host is on a different domain.

**Comment by jmalonzo** [comment]
Thanks. That should be fixed now.

**Comment by erikringsmuth** [comment]
Cool! I'll build a minified version and tag a new release tonight.

**Comment by jmalonzo** [comment]
:+1:

**Comment by erikringsmuth** [comment]
Tag pushed as `v0.3.1` :smile:

**Comment by jmalonzo** [comment]
Hi Erik

I think I might have broken Safari here - looks like startsWith isn't
supported there. Perhaps we can use indexOf/lastIndexOf? I'm away atm so
can't create a PR atm. Sorry

**Comment by erikringsmuth** [comment]
I think we can change it to `href.indexOf('http') === 0`. It should do the same thing.

**Comment by jmalonzo** [comment]
Thanks. I jist raised a PR. Cheers

**Pull request merged**